File: src/main/java/com/forcetracker333/service/RiskAssessmentService.java
package com.forcetracker333.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.forcetracker333.domain.RiskAssessment;
import com.forcetracker333.dto.RiskAssessmentDTO;
import com.forcetracker333.dto.RiskAssessmentSearchDTO;
import com.forcetracker333.dto.RiskAssessmentPageDTO;
import com.forcetracker333.dto.RiskAssessmentConvertCriteriaDTO;
import com.forcetracker333.service.GenericService;
import com.forcetracker333.dto.common.RequestDTO;
import com.forcetracker333.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface RiskAssessmentService extends GenericService<RiskAssessment, Integer> {

	List<RiskAssessment> findAll();

	ResultDTO addRiskAssessment(RiskAssessmentDTO riskAssessmentDTO, RequestDTO requestDTO);

	ResultDTO updateRiskAssessment(RiskAssessmentDTO riskAssessmentDTO, RequestDTO requestDTO);

    Page<RiskAssessment> getAllRiskAssessments(Pageable pageable);

    Page<RiskAssessment> getAllRiskAssessments(Specification<RiskAssessment> spec, Pageable pageable);

	ResponseEntity<RiskAssessmentPageDTO> getRiskAssessments(RiskAssessmentSearchDTO riskAssessmentSearchDTO);
	
	List<RiskAssessmentDTO> convertRiskAssessmentsToRiskAssessmentDTOs(List<RiskAssessment> riskAssessments, RiskAssessmentConvertCriteriaDTO convertCriteria);

	RiskAssessmentDTO getRiskAssessmentDTOById(Integer riskAssessmentId);







}

File: forcetracker333_ui/src/services/RiskAssessmentService.js
import http from "../http-common";

class RiskAssessmentService {
  getAllRiskAssessments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/riskAssessment/riskAssessments`, searchDTO);
  }

  get(riskAssessmentId) {
    return this.getRequest(`/riskAssessment/${riskAssessmentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/riskAssessment?field=${matchData}`, null);
  }

  addRiskAssessment(data) {
    return http.post("/riskAssessment/addRiskAssessment", data);
  }

  update(data) {
  	return http.post("/riskAssessment/updateRiskAssessment", data);
  }
  
  uploadImage(data,riskAssessmentId) {
  	return http.postForm("/riskAssessment/uploadImage/"+riskAssessmentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new RiskAssessmentService();
